corehq/motech/repeaters/tests/test_payload_generators.py

```diff
@@ -0,0 +1,184 @@
+        """
+        When an integer is taken from form JSON or a case block, it is a
+        ``str``. When it is taken from a case property, it is still a ``str``.
+        """
+        gen = FormRepeaterJsonPayloadGenerator(None)
+        form_dict, info = self.get_payload_info(gen)
+
+        self.assertIsInstance(form_dict['form']['year'], str)
+        self.assertIsInstance(info.form_question_values['/data/year'], str)
+        self.assertIsInstance(info.updates['year'], str)
+        self.assertIsInstance(info.extra_fields['year'], str)  # <--
+
+    def test_form_dict_integer(self):
+        """
+        When an integer is taken from form JSON or a case block, it is a
+        ``str``. When it is taken from a case property, it is still a ``str``.
+        """
+        gen = FormDictPayloadGenerator(None)
+        form_dict, info = self.get_payload_info(gen)
+
+        self.assertIsInstance(form_dict['form']['year'], str)
+        self.assertIsInstance(info.form_question_values['/data/year'], str)
+        self.assertIsInstance(info.updates['year'], str)
+        self.assertIsInstance(info.extra_fields['year'], str)  # <--
+
+    def test_form_json_multiplechoice(self):
+        gen = FormRepeaterJsonPayloadGenerator(None)
+        form_dict, info = self.get_payload_info(gen)
+
+        self.assertIsInstance(form_dict['form']['breakfast'], str)
+        self.assertIsInstance(info.form_question_values['/data/breakfast'], str)
+        self.assertIsInstance(info.updates['breakfast'], str)
+        self.assertIsInstance(info.extra_fields['breakfast'], str)
+
+    def test_form_dict_multiplechoice(self):
+        gen = FormDictPayloadGenerator(None)
+        form_dict, info = self.get_payload_info(gen)
+
+        self.assertIsInstance(form_dict['form']['breakfast'], str)
+        self.assertIsInstance(info.form_question_values['/data/breakfast'], str)
+        self.assertIsInstance(info.updates['breakfast'], str)
+        self.assertIsInstance(info.extra_fields['breakfast'], str)
+
+    def get_payload_info(
+        self,
+        payload_generator: BasePayloadGenerator,
+    ) -> Tuple[dict, CaseTriggerInfo]:
+        payload = payload_generator.get_payload(None, self.form)
+        if isinstance(payload, str):
+            payload = json.loads(payload)
+        [info] = get_relevant_case_updates_from_form_json(
+            DOMAIN,
+            payload,
+            case_types=None,
+            extra_fields=['year', 'breakfast', 'decimal'],
+            form_question_values=get_form_question_values(payload),
+        )
+        return payload, info
+
+
+def post_xform(xform_id, case_id):
+    user_id = uuid4().hex
+    now_str = datetime.utcnow().isoformat(timespec='milliseconds') + 'Z'
+    xform = f"""<?xml version='1.0' ?>
+<data xmlns="https://commcarehq.org/test/TestDataTypes/">
+
+    <name>spam</name>
+    <year>1970</year>
+    <breakfast>spam egg spam spam bacon spam</breakfast>
+    <decimal>4.4</decimal>
```

Comment:
:eyes:

Reply:
This is totally unexpected to me. If you change the name of this case property, does it do something different?

Reply:
Sorry, I didn't notice your question before merging. I should have updated the test. No, the property name does not affect the data type.

This seems to go way back. Here is [a comment from 2013](https://github.com/dimagi/commcare-hq/commit/c65823f04445e8ac2bb8c8aec3dd7a3c072e01bd#diff-9ecea39090f97daa0a6554553e2897c15f6de14fdfec8cda8e87e00dfe20dc93R228-R231) about "couchdbkit's auto-type detection".